### PR TITLE
Integrate k-NN functionality with security plugin

### LINKED ISSUE #2265: Onboard KNN Plugin features into security
Related to #2256

We would like to integrate the k-NN plugin functionality with security. 

For k-NN, we have several APIs. Listed below are the apis, the roles we want to assign the apis (and there all of the actions) and the actions:
1. SearchModel (role: knn_read_access)
cluster:admin/knn_search_model_action
2. TrainModel (role: knn_full_access)
cluster:admin/knn_training_model_action
cluster:admin/knn_training_job_router_action
cluster:admin/knn_training_job_route_decision_info_action
3. Warmup (role: knn_full_access)
cluster:admin/knn_warmup_action
4. DeleteModel (role: knn_full_access)
cluster:admin/knn_delete_model_action
cluster:admin/knn_remove_model_from_cache_action
cluster:admin/knn_update_model_graveyard_action
5. GetModel (role: knn_read_access)
cluster:admin/knn_get_model_action
6. Stats (role: knn_read_access)
cluster:admin/knn_stats_action

In order to create these roles, do I just need to update:
1. https://github.com/opensearch-project/security/blob/main/config/roles.yml
2. https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/configuration/constants.tsx#L32

In addition, we have 1 system index: ".opensearch-knn-models". Is there anything we should do in particular in security plugin for this? Do we need to add it to a sample config anywhere?

### PR DESCRIPTION
**Pull request by jmazanec15**
### Description
Adds 2 k-NN related roles to the plugin: read only and fully access.

Adds k-NN system index to demo scripts as well.

### Issues Resolved
#2265 

### Testing
Pending

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peternied — APPROVED** [review]
Looks good to me, was there more work you wanted to do before merging?  I'd rather merge this change and be able to employ the distribution level automation for faster verification.

**Review by DarshitChanpura — APPROVED** [review]
LGTM! @jmazanec15 if there is nothing more to be added from your end, can you please mark this as ready for review and we will merge it.

**Comment by jmazanec15** [comment]
@peternied @DarshitChanpura Ready for review!

**Pull request merged**
